### PR TITLE
use imports instead of constants, fix various warnings output on js console

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Switches dependency references in various files to use `import` in favor of defining constants referencing the `wp.` global variables loaded in the editor. This ensures that our script registration can define the proper dependencies.

Additionally fixes several (minor) issues, mostly related to deprecations from the WP core (@wordpress/* packages)

Also replaces use of `class=` with `className=` in various JSX templates. Use of `class` throws warnings but is properly interpreted. This fix prevents warnings from being thrown in the console.

## How has this been tested?

Manually tested affected blocks and components.

## Types of changes

Bug fixes

There's a chance that switching from `wp.components.ServerSideRender` to `@wordpress/server-side-render` could cause backwards compatibility issues for anyone using old versions of the Gutenberg plugin or WordPress. I'm not sure the best way to import something in a backwards compatible way.

## Checklist:

- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: assets/src/blocks/course-information/index.js [hunk 1]
```diff
@@ -6,6 +6,7 @@
  * @since [version] Remove import of empty CSS file.
  *              Fix UX issues when all information options are disabled.
  *              Use import instead of require.
+ *              Import RichText from @wordpress/blockEditor in favor of @wordpress/editor
```
Comment (eri-trabiccolo) [inline assets/src/blocks/course-information/index.js:9]:
```suggestion
 *              Import RichText from @wordpress/blockEditor in favor of @wordpress/editor.
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
This is good to me, tested, works fine.

I could even not request any changes but I wanted to grab your attention on the min WP version required to use the `@wordpress/block-editor`. I might be totally wrong on this though.

File: assets/src/blocks/course-information/index.js [hunk 2]
```diff
@@ -18,7 +19,7 @@ import Inspector from './inspect'
 import PreviewTerms from './preview-terms'
 
 // External Deps.
-import { RichText } from '@wordpress/editor';
+import { RichText } from '@wordpress/block-editor';
```
[the remark this answers is not in the document]
Comment (thomasplevy) [reply assets/src/blocks/course-information/index.js:22]: I was making some assumptions that this wouldn't cause any version-raising issues and I didn't want to think about it too deeply but this is an important point.

Thankfully @wordpress/scripts handles this backwards compatibility quite nicely. I had a look at the generated js file (the unminified version) and here's what is generated by using import included in this PR:

```js
var _ref = wp.blockEditor || wp.editor, 
    RichText = _ref.RichText,
    __ = wp.i18n.__,
    _wp$element = wp.element,
    RawHTML = _wp$element.RawHTML,
    Component = _wp$element.Component,
    Fragment = _wp$element.Fragment;
```

So this means that *no* we do not have to raise our supported wp to 5.2.

Nice job WordPress (not sarcastic)

[the remark this answers is not in the document]
Comment (eri-trabiccolo) [reply assets/src/blocks/course-information/index.js:22]: That's really cool, and you switched to @wordpress/scripts pretty recently right?

**Pull request merged**